src/test/java/io/github/josiasbarreto_dev/desafio_naruto/controller/CharacterControllerTest.java

```diff
@@ -0,0 +1,231 @@
+package io.github.josiasbarreto_dev.desafio_naruto.controller;
+
+import io.github.josiasbarreto_dev.desafio_naruto.dto.ChakraRequestDTO;
+import io.github.josiasbarreto_dev.desafio_naruto.dto.CharacterRequestDTO;
+import io.github.josiasbarreto_dev.desafio_naruto.dto.CharacterResponseDTO;
+import io.github.josiasbarreto_dev.desafio_naruto.dto.JutsuRequestDTO;
+import io.github.josiasbarreto_dev.desafio_naruto.model.NinjaType;
+import io.github.josiasbarreto_dev.desafio_naruto.service.CharacterService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+@DisplayName("Controller Unit Tests")
+public class CharacterControllerTest {
+    @InjectMocks
+    private CharacterController characterController;
+
+    @Mock
+    private CharacterService characterService;
+
+    private ChakraRequestDTO payloadChakraRequestDTO;
+    private CharacterRequestDTO payloadRequestDTO;
+    private CharacterRequestDTO payloadRequestWithNinjaTypeInvalid;
+    private JutsuRequestDTO payloadJutsuRequestDTO;
+    private CharacterResponseDTO payloadResponseDTO;
+    private CharacterResponseDTO payloadJutsuResponseDTO;
+    private CharacterResponseDTO payloadChakraResponseDTO;
+
+    private final Long VALID_CHARACTER_ID = 1L;
+    private final NinjaType NINJA_TYPE = NinjaType.TAIJUTSU;
```

Comment:
Poderia deixar o nome mais claro sobre o tipo de ninja
```suggestion
    private final NinjaType NINJA_TYPE_TAIJUTSU = NinjaType.TAIJUTSU;
```